src/main/java/training/endava/skeleton/Classes/Person.java

```diff
@@ -0,0 +1,124 @@
+package training.endava.skeleton.Classes;
+
+import training.endava.skeleton.Exceptions.IncorectRuntimeException;
+import training.endava.skeleton.Interfaces.SkeletonObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Person implements SkeletonObject<Person, Integer> {
+
+    private Integer ID;
+    private String name;
+    private String phoneNumber;
+    private String adress;
+    private static List<Integer> IDs;
+
+    public Person(Integer ID, String name, String phoneNumber, String adress) {
+        if (IDs == null){
+            IDs = new ArrayList<>();
+        }
+        checkAll(ID, name, phoneNumber, adress);
+
+        this.ID = ID;
+        this.IDs.add(ID);
+        this.name = name;
+        this.phoneNumber = phoneNumber;
+        this.adress = adress;
+    }
+
+    @Override
+    public Integer getId() {
+        return this.ID;
+    }
+
+    @Override
+    public void setId(Integer integer) {
+        this.IDs.remove(this.ID);
+        this.IDs.add(integer);
+        this.ID = integer;
```

Comment:
You don't need this remove and add. Just use this.id = integer. A good practice is to use id also for the setter argument.